sycl/source/detail/helpers.cpp

```diff
@@ -54,6 +66,47 @@ void waitEvents(std::vector<sycl::event> DepEvents) {
   }
 }
 
+std::vector<RT::PiEvent>
+Command::getPiEvents(const std::vector<EventImplPtr> &EventImpls) const {
```

Comment:
Would it make sense to create a separate function for this case instead of modifying `Command::getPiEvents`? In all other cases we expect these to already be enqueued so this logic is exclusive to barriers, and it's not obvious that a function called `getPiEvents` might enqueue anything.